src/firebase/auth/index.ts

```diff
@@ -25,11 +25,13 @@ export const auth = getAuth(app)
  */
 export const createNewUser = async ({ firstName, lastName, email, password }: INewUserProps): Promise<void> => {
 	try {
-		const newUser = await createUserWithEmailAndPassword(auth, email as string, password as string)
-
-		await updateProfile(newUser.user, {
-			displayName: `${firstName} ${lastName}`,
-		})
+		if (typeof email === 'string' && typeof password === 'string') {
```

Comment:
it's better to do something like:
```
if (typeof email === 'string' && typeof password === 'string') throw new TypeError("email and/or password is not a string")
```
before the try catch block
and then start a try block without nested if statemebt.